.github/workflows/build.yml

```diff
@@ -0,0 +1,336 @@
+        uses: SonarSource/ci-github-actions/cache@v1
+        with:
+          path: ${{ env.ORCHESTRATOR_HOME }}
+          key: orchestrator-${{ env.TODAY }}
+          enableCrossOsArchive: true
+      - uses: SonarSource/vault-action-wrapper@v3
+        id: secrets
+        with:
+          secrets: |
+            development/artifactory/token/{REPO_OWNER_NAME_DASH}-public-reader access_token | ARTIFACTORY_ACCESS_TOKEN;
+            development/github/token/licenses-ro token | GITHUB_TOKEN;
+      - name: Run Integration Tests
+        env:
+          ARTIFACTORY_ACCESS_TOKEN: ${{ fromJSON(steps.secrets.outputs.vault).ARTIFACTORY_ACCESS_TOKEN }}
+          GITHUB_TOKEN: ${{ fromJSON(steps.secrets.outputs.vault).GITHUB_TOKEN }}
+          GRADLE_TASK: its:plugin:integrationTest
+          KEEP_ORCHESTRATOR_RUNNING: "true"
+          SQ_VERSION: ${{ matrix.SQ_VERSION }}
+        run: |
+          ./gradlew "${GRADLE_TASK}" "-Dsonar.runtimeVersion=${SQ_VERSION}" --info --build-cache --console plain --no-daemon
+      - name: Upload test results
+        if: always() && ! cancelled()
+        uses: actions/upload-artifact@330a01c490aca151604b8cf639adc76d48f6c5d4 # v5.0.0
+        with:
+          name: qa-plugin-test-results-${{ matrix.SQ_VERSION }}
+          path: '**/test-results/**/*.xml'
+      - name: Upload reports
+        if: always() && ! cancelled()
+        uses: actions/upload-artifact@330a01c490aca151604b8cf639adc76d48f6c5d4 # v5.0.0
+        with:
+          name: qa-plugin-reports-${{ matrix.SQ_VERSION }}
+          path: '**/build/reports/**/*'
+
+  qa_ruling:
+    needs: [build]
+    runs-on: sonar-l-public
+    name: QA Ruling
+    if: *skip-qa-jobs
+    permissions:
+      id-token: write
+      contents: write
+    strategy:
+      matrix:
+        PHP_PROJECT: [Flysystem, Monica, PhpCodeSniffer, PhpMailer, Psysh, PhpWord, RubixML, PhpSpreadsheet]
+    steps:
+      - uses: actions/checkout@93cb6efe18208431cddfb8368fd83d5badbf9bfd # v5.0.1
+        with:
+          submodules: true
+      - uses: jdx/mise-action@9dc7d5dd454262207dea3ab5a06a3df6afc8ff26 # v3.4.1
+        with:
+          version: 2025.7.12
+      - *create-gradle-user-home
+      - *cache-gradle-dependencies
+      - *setup-orchestator-home
+      - *cache-orchestrator
+      - uses: SonarSource/vault-action-wrapper@v3
+        id: secrets
+        with:
+          secrets: |
+            development/artifactory/token/{REPO_OWNER_NAME_DASH}-public-reader access_token | ARTIFACTORY_ACCESS_TOKEN;
+            development/github/token/licenses-ro token | GITHUB_TOKEN;
+      - name: Run Ruling Integration Tests
+        env:
+          ARTIFACTORY_ACCESS_TOKEN: ${{ fromJSON(steps.secrets.outputs.vault).ARTIFACTORY_ACCESS_TOKEN }}
+          GITHUB_TOKEN: ${{ fromJSON(steps.secrets.outputs.vault).GITHUB_TOKEN }}
+          GRADLE_TASK: its:ruling:integrationTest
+          KEEP_ORCHESTRATOR_RUNNING: "true"
+          SQ_VERSION: 2025.4.3.113915
```

Comment:
How come here we use this version, while in the `qa plugin` we use `25.7.0.110598`?